apps/blockchain/test/blockchain/block_test.exs

```diff
@@ -337,11 +337,22 @@ defmodule Blockchain.BlockTest do
 
       assert Enum.count(block.transactions) == 1
 
-      expected_receipt = %Transaction.Receipt{
-        bloom_filter: "",
+      expected_receipt = %Blockchain.Transaction.Receipt{
+        bloom_filter:
+          <<0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0,
+            0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0, 0>>,
         cumulative_gas: 28_180,
         logs: [],
-        state: block.header.state_root
+        state:
+          <<61, 115, 119, 182, 91, 209, 64, 217, 127, 241, 158, 23, 15, 12, 238, 234, 204, 16, 1,
```

Comment:
Why is this no longer equal to `block.header.state_root`? If it is now equal to some other value, could we make that the expectation? The reason why I bring this point is that when we use raw values like this one in tests, and someone else comes looking later, it is really hard to understand what this test is expecting. Clearly, it expects this binary to be there. But why this binary and where did it come from?

Reply:
@germsvel Actually, it still equals to `block.header.state_root`. My bad